File: Java-SpringBoot/BE-01-InMemory/java-17-springboot-3.5/src/main/java/com/flyrank/userservice/service/UserService.java
package com.flyrank.userservice.service;

import com.flyrank.userservice.dto.UserCreateRequest;
import com.flyrank.userservice.dto.UserResponse;

import java.util.UUID;

/**
 * Service layer interface defining business operations for Users.
 */
public interface UserService {
    /**
     * Creates a new user with a generated UUID.
     */
    UserResponse createUser(UserCreateRequest request);

    /**
     * Retrieves an existing user by ID or throws UserNotFoundException.
     */
    UserResponse getUserById(UUID id);
}

File: Java-SpringBoot/BE-01-InMemory/kotlin-2.3-springboot-3.5/src/main/kotlin/com/flyrank/userservice/service/UserService.kt
package com.flyrank.userservice.service

import com.flyrank.userservice.dto.UserCreateRequest
import com.flyrank.userservice.dto.UserResponse
import java.util.UUID

/**
 * Service layer interface in Kotlin.
 */
interface UserService {
    fun createUser(request: UserCreateRequest): UserResponse
    fun getUserById(id: UUID): UserResponse
}
